Automattic/jetpack
# My Jetpack: Show a Boost score Tooltip in the Boost product card.

This PR Adds the Boost score Tooltip that is shown when hovering or focusing on the Boost score letter grade (See image ⤵️).

![Screenshot 2024-02-16 at 11 51 58 AM](https://github.com/Automattic/jetpack/assets/11078128/e3d16f9e-7de0-4e30-a8af-0051c26aed68)






Note: **Still TODO** (in a separate follow-up PR):
- Display different Tooltip content, based on whether the plugin is installed or not and whether the user owns a paid Boost plan or not.  See Boost Tooltip Google Doc: 33a29-pb

## Proposed changes:

* Added a tsconfig.json file to the my-jetpack package for improved typescript compatibility.  (prior to this, the types in global.d.ts were not fully being loaded and recognized in VSCode). Now fixed.
* Added a BoostScoreTooltip component
* Import and display the new BoostScoreTooltip component when hovering over the Boost score letter-grade.

### Other information:


## Jetpack product discussion



Related to: [Add value to the active Boost product card on My Jetpack](https://github.com/orgs/Automattic/projects/702/views/6?pane=issue&itemId=50279848)

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout this branch using the Jetpack Beta plugin, Or using your local dev environment.
* Connect Jetpack ( at least site connection), so that the Boost speed scores will show in the My Jetpack Boost product card).
* Go To My Jetpack page.
* In the Boost product card, hover your mouse over the Boost score "letter grade".  Verify that the Boost score Tooltip appears and that it looks visually similar to the Figma design: e7y1h9DsyIG8XQ4a9mUT6P-fi-4411%3A18538
* Use your keyboard, tab into the Boost score letter-grade and verify the Tooltip opens when focused on using the keyboard.
* Check that the Tooltip works and looks ok in mobile view.
